tests/shared/core/training_data/story_reader/test_common_story_reader.py

```diff
@@ -297,3 +297,42 @@ async def test_yaml_slot_different_types(stories_file: Text, domain: Domain):
     assert tracker[0].events[3] == SlotSet(key="list_slot", value=["value1", "value2"])
     assert tracker[0].events[4] == SlotSet(key="bool_slot", value=True)
     assert tracker[0].events[5] == SlotSet(key="text_slot", value="some_text")
+
+
+async def test_can_read_test_story_with_or_slot_was_set_statement(tmp_path: Path, domain: Domain):
+    story_with_or_slot_was_set = """
+    stories:
+    - story: simple_story_with_or_slot_was_set
+      steps:
+      - intent: simple
+      - action: utter_default
+      - or:
+        - slot_was_set:
+            - name: joe
+        - slot_was_set:
+            - name: bob
+      - action: utter_greet
+    """
+    stories_path = tmp_path / "stories.yml"
+    stories_path.write_text(story_with_or_slot_was_set, "utf8")
+
+    trackers = await training.load_data(
+        str(stories_path),
+        domain,
+        use_story_concatenation=False,
+        tracker_limit=1000,
+        remove_duplicates=False,
+    )
+    tracker = trackers[0]
+    assert tracker.events[0] == ActionExecuted("action_listen")
+    assert tracker.events[1] == UserUttered(
+        intent={INTENT_NAME_KEY: "simple", "confidence": 1.0},
+        parse_data={
+            "text": "/simple",
+            "intent_ranking": [{"confidence": 1.0, INTENT_NAME_KEY: "simple"}],
+            "intent": {"confidence": 1.0, INTENT_NAME_KEY: "simple"},
+            "entities": [],
+        },
+    )
+    assert tracker.events[2] == ActionExecuted("utter_default")
+    assert tracker.events[3] == SlotSet(key="name", value="joe")
```

Comment:
There's a missing newline at the end here.
Also, would index 4 in `tracker.events` also be a `SlotSet` event, the one with `value = bob`?

Reply:
That's a good assumption but no — however, I am not sure why. My assumption was that there should be two trackers and `trackers[0].events[3] == SlotSet(key="name", value="joe")` and `trackers[1].events[3] == SlotSet(key="name", value="bob")` but that doesn't happen.
I was thinking that's an issue but it doesn't happen with or expressions when the intents (not `SlotSet`s) are used, and it was like that before all these changes — so I now wonder if it's the bug or I just got that all wrong.
Maybe somebody can clarify? @joejuzl or @wochinge or maybe even someone from research

Reply:
Very good catch @ancalita ! It should be two trackers - one with "joe" and one with "bob". If you do the `or` with intents you also get two trackers. This might require some changes in the `TrackerGenerator`

Reply:
This still needs to be handled right @alwx ?

Reply:
Should be good to go now — I rebased the PR: there is no `common_story_reader.py` anymore, and the test to check if story steps get copied or not has already been added to `test_yaml_story_reader.py` (e.g. `test_or_statement_with_slot_was_set`).

Reply:
I think the rebasing just deleted this test. We still need a test which asserts that the generator creates two trackers out of the story with the `or` slot. So you basically need a test which asserts that `training.load_data` causes 2 trackers

Reply:
Finally added a test for this! (`test_or_statement_story_with_or_slot_was_set`)